contentprovider.go

```diff
@@ -588,6 +588,22 @@ func findMaxOverlappingSection(secs []DocumentSection, off, sz uint32) (uint32,
 	return uint32(j), ol1 > 0
 }
 
+func (p *contentProvider) matchesSymbol(cm *candidateMatch) bool {
```

Comment:
We are duplicating some checks, since we run both `calculateTermFrequency` for the overall file score, plus `candidateMatchScore` for the individual chunk scores. It would be good to unify these, but I didn't want to embark on a big refactor in this PR.